kit/controls/src/UIIndicator.tsx

```diff
@@ -11,20 +20,80 @@ export type UIIndicatorProps = {
     style?: StyleProp<ViewStyle>;
 };
 
+/**
+ * The maximum length of the visible arc relative to the circumference
+ */
+// @inline
+const MAXIMUM_DASH_LENGTH_COEFFICIENT = 0.8;
+
+export const AnimatedCircle = Animated.createAnimatedComponent(
+    addNativeProps(Circle, {
+        strokeDashoffset: true,
+        /**
+         * Used for web
+         */
+        'stroke-dashoffset': true,
+    }),
+);
+
 export const UIIndicator = React.memo(function UIIndicator({
     color = ColorVariants.LineAccent,
-    size,
-    trackWidth,
+    size = UIConstant.indicator.defaultSize,
+    trackWidth = UIConstant.indicator.defaultTrackWidth,
     style,
 }: UIIndicatorProps) {
     const theme = useTheme();
 
+    /**
+     * Progress changes from 0 to 1 continually repeating
+     */
+    const progress = useSharedValue<number>(0);
+
+    React.useEffect(() => {
+        progress.value = withRepeat(withTiming(1, { duration: 2000, easing: Easing.linear }), -1);
+    }, [progress]);
+
+    const animatedStyle = useAnimatedStyle(() => {
+        return {
+            transform: [
+                {
+                    rotate: `${progress.value * 2 * Math.PI}rad`,
+                },
+            ],
+        };
+    }, []);
+
+    const circumference = React.useMemo(() => size * Math.PI, [size]);
+
+    const animatedProps = useAnimatedProps(() => {
+        const strokeDashoffset =
+            circumference * (1 - progress.value * 2) * MAXIMUM_DASH_LENGTH_COEFFICIENT;
```

Comment:
Is it OK that it can become negative?

Reply:
Yes, it should change from `circumference * MAXIMUM_DASH_LENGTH_COEFFICIENT` to `-circumference * MAXIMUM_DASH_LENGTH_COEFFICIENT`